src/fastcs_odin/frame_receiver.py

```diff
@@ -1,17 +1,29 @@
-from fastcs_odin.odin_adapter_controller import (
-    OdinAdapterController,
+from fastcs_odin.odin_data import OdinDataAdapterController
+from fastcs_odin.odin_subcontroller import OdinSubController
+from fastcs_odin.util import (
+    OdinParameter,
+    create_attribute,
+    partition,
+    remove_metadata_fields_paths,
 )
-from fastcs_odin.odin_data import OdinDataAdapterController, OdinDataController
-from fastcs_odin.util import OdinParameter, partition
 
 
-class FrameReceiverController(OdinDataController):
+class FrameReceiverController(OdinSubController):
     async def initialise(self):
-        self._process_parameters()
-
         def __decoder_parameter(parameter: OdinParameter):
             return "decoder" in parameter.path[:-1]
 
+        for parameter in self.parameters:
+            # Remove duplicate index from uri
+            parameter.uri = parameter.uri[1:]
+            # Remove redundant status/config from parameter path
+            parameter.set_path(parameter.uri[1:])
+            self.add_attribute(
+                parameter.name,
+                create_attribute(parameter=parameter, api_prefix=self._api_prefix),
+            )
+        self.parameters = remove_metadata_fields_paths(self.parameters)
```

Comment:
This isn't right. `FrameReceiverController` is creating a `decoder/name` attribute again, which fails to update. We need to do

1. remove_metadata_fields_paths
2. partition decoder parameters
3. create_attributes